### PR TITLE
Updating setter to downcase Stem Course Template No

### PR DESCRIPTION
**Pull request by msquance-stem**
Adding test cases and updating seed
Updates need to be done in the console before deploying this, to prevent courses being double seeded

## Status

* Current Status: Ready for tech review
* Closes https://github.com/NCCE/teachcomputing.org-issues/issues/2628

## Review progress:

- [ ] Tech review completed

## What's changed?

- Making the setter automatically downcase the ID
- Updated the seed file

## Steps to perform before deploying to production

The ID's listed in the issue need to be downcased **before** deployment to production otherwise the seed will generate a second copy of the course.

### CODE REVIEW

File: app/models/activity.rb [hunk 1]
```diff
@@ -53,6 +53,11 @@ def self.cs_accelerator_diagnostic_tool
     end
   end
 
+  def stem_course_template_no= value
+    # Ensure that all stem_course_template_no's are stored in downcase format
+    super(value.downcase)
+  end
+
```
Comment (abr-storm) [inline app/models/activity.rb:60]: It's a shame we're not on 7.1 yet, we could have used the new normalize feature here https://www.bigbinary.com/blog/rails-7-1-adds-activerecord-base-normalizes

**Review by abr-storm — APPROVED** [review]

File: app/models/activity.rb (hunk 1 above)
Comment (Caleb-T-Owens) [reply app/models/activity.rb:60]: @msquance-stem could you please use brackets around your setter's arguments?

`def stem_course_template_no=(value)`

**Review by Caleb-T-Owens — CHANGES_REQUESTED** [review]
One style comment

**Review by Caleb-T-Owens — APPROVED** [review]
LGTM

**Pull request merged**